lib/tasks/omdb.rake

```diff
@@ -0,0 +1,250 @@
+    countries.each do |country|
+      new_country = MovieCountry.new
+      new_country.country = country
+      new_country.movie_id = movie_id
+
+      new_country.save
+    end
+  end
+
+  # Method to save directors to the database
+  def save_directors(directors, movie_id)
+    # Saves the list of directors for a given movie to the database
+    # Params:
+    # - directors: Array of director names
+    # - movie_id: IMDB ID of the movie
+
+    puts "Saving Directors..."
+    directors.each do |director|
+      new_director = MovieDirector.new
+      new_director.name = director
+      new_director.movie_id = movie_id
+
+      new_director.save
+    end
+  end
+
+  # Method to save genres to the database
+  def save_genres(genres, movie_id)
+    # Saves the list of genres for a given movie to the database
+    # Params:
+    # - genres: Array of genre names
+    # - movie_id: IMDB ID of the movie
+
+    puts "Saving Genres..."
+    genres.each do |genre|
+      new_genre = MovieGenre.new
+      new_genre.genre = genre
+      new_genre.movie_id = movie_id
+
+      new_genre.save
+    end
+  end
+
+  def save_languages(languages, movie_id)
+    # Saves the list of languages for a given movie to the database
+    # Params:
+    # - languages: Array of language names
+    # - movie_id: IMDB ID of the movie
+
+    puts "Saving Languages..."
+    languages.each do |language|
+      new_language = MovieLanguage.new
+      new_language.language = language
+      new_language.movie_id = movie_id
+
+      new_language.save
+    end
+  end
+
+  def save_ratings(ratings, movie_id)
+    # Saves the list of ratings for a given movie to the database
+    # Params:
+    # - ratings: Array of rating details (Source and Value)
+    # - movie_id: IMDB ID of the movie
+
+    puts "Saving Ratings..."
+    ratings.each do |rating|
+      new_rating = MovieRating.new
+      new_rating.source = rating['Source']
+      new_rating.value = rating['Value']
+      new_rating.movie_id = movie_id
+
+      new_rating.save
+    end
+  end
+
+  # Method to save writers to the database
+  def save_writers(writers, movie_id)
+    # Saves the list of writers for a given movie to the database
+    # Params:
+    # - writers: Array of writer names
+    # - movie_id: IMDB ID of the movie
+
+    puts "Saving Writers..."
+    writers.each do |writer|
+      new_writer = MovieWriter.new
+      new_writer.name = writer
+      new_writer.movie_id = movie_id
+
+      new_writer.save
+    end
+  end
+
+  # Method to fetch movie details from the OMDB API
+  def fetch_movie_details(imdb_id)
+    # Fetches movie details from the OMDB API using the provided IMDB ID
+    # Params:
+    # - imdb_id: IMDB ID of the movie to fetch details for
+    # Returns: Hash containing movie details
+
+    api_url = ENV['api_url']
+    url_params = URI.encode_www_form(i: imdb_id)
+
+    url = "#{api_url}&#{url_params}"
+    uri = URI.parse(url)
+    response = Net::HTTP.get_response(uri)
```

Comment:
Add a check here like if the API is returning 200 or not, then only you'll return response, otherwise you need to handle the retry logic or restart the task.